crates/utils/src/utils/markdown/mod.rs

```diff
@@ -98,6 +99,33 @@ pub fn markdown_rewrite_image_links(mut src: String) -> (String, Vec<Url>) {
   (src, links)
 }
 
+pub fn markdown_check_links(text: &str, blocklist: Vec<String>) -> LemmyResult<()> {
+  for blocked_url in &blocklist {
+    let blocked = Url::parse(blocked_url)?;
+    let block_regex = if blocked_url.ends_with('/') {
+      Regex::new(&format!(
+        "({}://)?{}{}?",
+        blocked.scheme(),
+        blocked.domain().expect("No domain."),
```

Comment:
you'll need to [regex-escape](https://docs.rs/regex/latest/regex/fn.escape.html) domain and path, otherwise, blocking e.g. `l.mmy.ml` will lead to blocking `lemmy.ml`.

Reply:
Done.